### PR TITLE
Add remaining RelativeTimeFormatter constructors

### PR DESCRIPTION
**Pull request by pdogr**
#802

### CODE REVIEW

File: experimental/relativetime/src/relativetime.rs [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+    );
+    constructor!(
+        try_new_narrow_month_unstable,
+        NarrowMonthRelativeTimeFormatDataV1Marker
+    );
+    constructor!(
+        try_new_narrow_quarter_unstable,
+        NarrowQuarterRelativeTimeFormatDataV1Marker
+    );
+    constructor!(
```
Comment (sffc) [inline experimental/relativetime/src/relativetime.rs:240]: Nit: Make sure these are covered by tests. I suggest adding a unit test block in this file where you call every function for one of the test locales (maybe just English) just to make sure they work.

File: ffi/diplomat/ffi_coverage/tests/missing_apis.txt [hunk 2]
```diff
@@ -40,3 +40,39 @@ icu::properties::sets::UnicodeSetDataBorrowed::contains#FnInStruct
 icu::properties::sets::UnicodeSetDataBorrowed::contains32#FnInStruct
 icu::properties::sets::UnicodeSetDataBorrowed::contains_char#FnInStruct
 icu::properties::sets::load_basic_emoji#Fn
+icu::relativetime::FormattedRelativeTime#Struct
```
[the remark this answers is not in the document]
Comment (pdogr) [reply ffi/diplomat/ffi_coverage/tests/missing_apis.txt]: Added.

File: experimental/relativetime/src/relativetime.rs (hunk 1 above)
Comment (pdogr) [reply experimental/relativetime/src/relativetime.rs:240]: Added tests that cover all constructors.

File: experimental/relativetime/tests/tests.rs [hunk 3]
```diff
@@ -0,0 +1,798 @@
...
+use icu_relativetime::{options::Numeric, RelativeTimeFormatter, RelativeTimeFormatterOptions};
+use writeable::assert_writeable_eq;
+
+macro_rules! generate_test {
+    ($test_name: ident, $constructor: ident, $options: expr, [$(($time: literal, $expected: literal)),+ $(,)?]) => {
+        #[test]
+        fn $test_name(){
+            let relative_time_formatter = RelativeTimeFormatter::$constructor(
+                &icu_testdata::unstable(),
+                &locale!("ar").into(),
```
Comment (robertbastian) [inline experimental/relativetime/tests/tests.rs:16]: Please use English so we can sanity check that there are no mixups anywhere.

**Review by robertbastian — CHANGES_REQUESTED** [review]

File: experimental/relativetime/tests/tests.rs (hunk 3 above)
Comment (pdogr) [reply experimental/relativetime/tests/tests.rs]: Added tests containing English locale too. "ar" locale had cases not covered by "en" hence using both.

**Review by robertbastian — APPROVED** [review]

**Review by sffc — APPROVED** [review]

**Pull request merged**